_shared/project/pyproject.toml

```diff
@@ -59,6 +59,104 @@ filterwarnings = [
+    "G",     # flake8-logging-format (enforce usage of `extra` in logging calls)
+    "INP",   # flake8-no-pep420 (checks for missing __init__.py files)
+    "PIE",   # flake8-pie (miscellaneous)
+    "T20",   # flake8-print (checks for print and pprint statements)
+    "PT",    # flake8-pytest-style (checks for common pytest style and consistency issues)
+    "RSE",   # flake8-raise (checks for issues with raising exceptions)
+    "RET",   # flake8-return (checks for issues with return values)
+    "SLOT",  # flake8-slots (requires __slots__ in subclasses of immutable types)
+    "SIM",   # flake8-simplify (lots of code simplification checks)
+    "TID",   # flake8-tidy-imports (checks for issues with imports)
+    "TC",    # flake8-type-checking (checks for type checking imports that aren't in TYPE_CHECKING blocks)
+    "ARG",   # flake8-unused-arguments (checks for unused arguments)
+    "PTH",   # flake8-use-pathlib (checks for cases with pathlib could be used but isn't)
+    "TD",    # flake8-todos (enforces good style for "# TODO" comments)
+    "FIX",   # flake8-fixme (checks for FIXMEs, TODOs, HACKs, etc)
+    "ERA",   # eradicate (checks for commented-out code)
+    "PGH",   # pygrep-hooks (miscellaneous)
+    "PL",    # pylint (miscellaneous rules from pylint)
+    "TRY",   # tryceratops (various try/except-related checks)
+    "FLY",   # flynt (checks for old-style %-formatted strings)
+    "PERF",  # perflint (checks for performance anti-patterns)
+    "FURB",  # refurb (various "refurbishing and modernizing" checks)
+    "DOC",   # pydoclint (docstring checks)
+    "RUF",   # Ruff-specific rules
+    "COM",   # flake8-commas (we used a code formatter so we don't need a linter to check this)
+    "D100","D101","D102","D103","D104","D105","D106","D107", # Missing docstrings.
+    "D202", # "No blank lines allowed after function docstring" conflicts with the Ruff code formatter.
+    # "Multi-line docstring summary should start at the first line" (D212)
+    # and "Multi-line docstring summary should start at the second line" (D213).
+    # These two rules conflict with each other so you have to disable one of them.
+    # How about we disable them both? PEP 257 says either approach is okay:
+    #
+    # > The summary line may be on the same line as the opening quotes or on
+    # > the next line.
+    # >
+    # > https://peps.python.org/pep-0257/#multi-line-docstrings
+    "D212", "D213",
+    "E501", # line-too-long (we use the code formatter so we don't need the linter to check line lengths for us).
+    "PLR2004", # "Magic value used in comparison", this mostly triggers false-positives related to HTTP status codes.
+    "PLR6301", # Method could be a function/classmethod/static method (doesn't use self)
+    "RET501", # Do not explicitly return None if it's the only possible return value.
+    "RET504", # Unnecessary assignment before return statement.
```

Comment:
I went for selecting ALL rules and then explicitly ignoring some. The idea is that we'll remove rules from this ignore list in time. This is a large ignore list for now to make it easier to migrate projects to Ruff, but I think we should aim to get this down to a pretty small list.